### PR TITLE
Fix for crashes during shutting down, mountpoints_mutex lock, sync the https port with the demos

### PR DESCRIPTION
**Pull request by marcin-sielski**

### CODE REVIEW

File: conf/janus.transport.http.cfg.sample.in [hunk 1]
```diff
@@ -15,7 +15,7 @@ port = 8088					; Web server HTTP port
 ;interface = eth0			; Whether we should bind this server to a specific interface only
 ;ip = 192.168.0.1			; Whether we should bind this server to a specific IP address (v4 or v6) only
 https = no					; Whether to enable HTTPS (default=no)
-;secure_port = 8889			; Web server HTTPS port, if enabled
+;secure_port = 8089			; Web server HTTPS port, if enabled
```
Comment (ploxiln) [inline conf/janus.transport.http.cfg.sample.in]: this looks like an unnecessary change for this PR

otherwise looks good

Comment (ploxiln) [inline conf/janus.transport.http.cfg.sample.in]: well, I see the commit for this change says it syncs it with the demos, but the default in the transport plugin is 8889:
https://github.com/meetecho/janus-gateway/blob/master/transports/janus_http.c#L683

File: janus.c [hunk 2]
```diff
@@ -3873,8 +3873,10 @@ gint main(int argc, char *argv[])
 	JANUS_LOG(LOG_INFO, "Destroying sessions...\n");
 	if(sessions != NULL)
 		g_hash_table_destroy(sessions);
+	sessions = NULL;
 	if(old_sessions != NULL)
 		g_hash_table_destroy(old_sessions);
+	old_sessions = NULL;
```
Comment (andreasg123) [inline janus.c]: I would put the assignments to `NULL` inside the `if` statements.  It won't make any noticeable performance difference but it is cleaner.

Comment (jswirl) [inline janus.c]: How about this:
`g_clear_pointer(&old_sessions, g_hash_table_destroy)`
Looks cleaner IMO.

Comment (andreasg123) [inline janus.c]: Yes, that does look cleaner.  The same pattern could be used for [this](https://github.com/MotorolaSolutions/janus-gateway/blob/1b9e4f0b164bb986c00463db3acabcef1d532928/janus.c#L3293):
`g_strfreev(list);`
`list = NULL;`
change to:
`g_clear_pointer(&list, g_strfreev);`

**Comment by marcin-sielski** [comment]
@ploxiln @andreasg123 Thank you for review. I have fixed that. Best Regards

**Comment by lminiero** [comment]
Thanks for this! :+1: As I said on the group, though, I'm on vacation now, so I'll only be able to review this in a couple of weeks when I get back.

File: janus.c [hunk 3]
```diff
@@ -3872,11 +3870,9 @@ gint main(int argc, char *argv[])
 
 	JANUS_LOG(LOG_INFO, "Destroying sessions...\n");
 	if(sessions != NULL)
-		g_hash_table_destroy(sessions);
-	sessions = NULL;
+		g_clear_pointer(&sessions, g_hash_table_destroy);
```
Comment (andreasg123) [inline janus.c]: You don't need the test `sessions != NULL` when using `g_clear_pointer`.

**Comment by marcin-sielski** [comment]
@andreasg123 fixed

**Comment by lminiero** [comment]
Thanks for this, looks good to me. And thanks for bringing `g_clear_pointer` to my attention, I was not aware of this method. I'll check if we can use it in other places too. Merging!
